src/test/java/com/google/sps/UserFoldersServletTest.java

```diff
@@ -0,0 +1,194 @@
+import static com.google.sps.tool.Tool.compareJson;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.gson.Gson;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.data.Folder;
+import java.util.List; 
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+
+
+@RunWith(JUnit4.class)
+public final class UserFoldersServletTest {
+
+  private final LocalServiceTestHelper helper = 
+    new LocalServiceTestHelper(
+      new LocalDatastoreServiceTestConfig()
+        .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+      new LocalUserServiceTestConfig())
+      .setEnvIsAdmin(true).setEnvIsLoggedIn(true)
+      .setEnvEmail("test@gmail.com").setEnvAuthDomain("gmail.com");
+
+  private static final Folder FOLDER_A = new Folder("FIRSTFOLDER", "en");
+  private static final Folder FOLDER_B = new Folder("SECONDFOLDER", "en");
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private UserFoldersServlet servlet;
+  private DatastoreService datastore;
+    
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    servlet = new UserFoldersServlet();
+    mockRequest = mock(HttpServletRequest.class);
+    mockResponse = mock(HttpServletResponse.class);
+    
+    // Set up a fake HTTP response 
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    // Initialize datastore
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    helper.tearDown();
+  }
+  
+  @Test
+  public void QueryUserFolders() throws Exception {
+    
+    // Generate testing User
```

Comment:
I still feel that comments can be more specific, like what is UserA and the fields it has. The test explanation's seem very generic and I'm not sure whats being tested.

Reply:
is the function name not clear enough to know what is occurring?

Reply:
Yeah, I'm not sure if this would be extra but what if you added whats going and what you expect. But this is kind of nit picky I guess.

Reply:
Made a commit for Change directory structure